tests/Silverback.Integration.Tests/Messaging/Validation/ValidatorConsumerBehaviorTests.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NSubstitute;
+using Silverback.Diagnostics;
+using Silverback.Messaging.Broker;
+using Silverback.Messaging.Broker.Behaviors;
+using Silverback.Messaging.Messages;
+using Silverback.Messaging.Sequences;
+using Silverback.Messaging.Validation;
+using Silverback.Tests.Types;
+using Silverback.Tests.Types.Domain;
+using Xunit;
+
+namespace Silverback.Tests.Integration.Messaging.Validation
+{
+    public class ValidatorConsumerBehaviorTests
+    {
+        [Fact]
+        public async Task HandleAsync_InvalidMessage_ExceptionIsThrown()
+        {
+            var endpoint = TestProducerEndpoint.GetDefault();
+            endpoint.MessageValidationMode = MessageValidationMode.LogWarning;
+            var headers = new[]
+            {
+                new MessageHeader("x-message-type", typeof(TestMessageValidation).AssemblyQualifiedName)
+            };
+
+            // TODO Chris: x-message-type seems to not work, the invalid message is valid
+            var envelope = new InboundEnvelope(
+                "{\"String10\": \"123456789abc\"}",
+                new MemoryStream(Encoding.UTF8.GetBytes("{\"String10\": \"123456789abc\"}")),
+                headers,
+                new TestOffset(),
+                new TestConsumerEndpoint("source-endpoint"),
+                "source-endpoint");
+
+            var logger = Substitute.For<IInboundLogger<ValidatorConsumerBehavior>>();
+
+            IRawInboundEnvelope? result = null;
+            await new ValidatorConsumerBehavior(logger).HandleAsync(
+                new ConsumerPipelineContext(
```

Comment:
I think there's an helper somewhere to create the context, iirc.

Reply:
@BEagle1984 I didn't found it 😐

Reply:
Here you go: https://github.com/BEagle1984/silverback/blob/master/tests/Silverback.Tests.Common.Integration/ConsumerPipelineContextHelper.cs#L22

It doesn't exist for the `ProducerPipelineContext` though. (You could create it if needed! 😃 )

Reply:
Thanks, I adjusted the tests using the helper.